### ISSUE TITLE
Maybe add "month" and "year" duration literals

### ISSUE THREAD

**Issue by webbedspace** [nushell/nushell#7435]
### Describe the solution you'd like

When using large durations (such as those obtained from subtracting filesizes) it can get awkward having to express years as `52wk` and months as `4wk`. See this example:
```
(date now) - $in |
      if $in < 1hr {
        '#e61919'
      } else if $in < 6hr {
        '#e68019'
      } else if $in < 1day {
        '#e5e619'
      } else if $in < 3day {
        '#80e619'
      } else if $in < 1wk {
        '#19e619'
      } else if $in < 4wk {
        '#19e5e6'
      } else if $in < 52wk {
        '#197fe6'
      } else { 'dark_gray' }
```

In this case, months and years would use the same assumption as `into duration` (1month = 30day, 1year = 365day). The only question is what form the suffixes should take. Currently, duration suffixes try to be less than 4 characters, so I guess these would use `1yr` and `1mon`… but maybe `mon` is a little too opaque as a suffix. While it could be `1month`, that'd break the established pattern, and `mth` is not really that obvious as being "month". Hrmm…

**Comment by fdncred** [comment]
There has been much discussion on this topic over the last ~3 years.

- jt added month and year
- jt removed them
- i added them + decades + centuries
- jt removed them again but still allowed the welcome banner to say something like 
```
It's been this long since Nushell's first commit:
3yr 7month 5day 22hr 1min 43sec 53ms 944µs
```

i believe @jntrnr's objection to these is that they're not accurate in all situations because months have different days, years can have leap years, etc. If that problem was resolved, it would probably be allowed again.

I've researched this quite a bit and I think it could be done but any date conversion or date math would have to include a begin datetime and and end datetime so that months and leaps could be calculated correctly. I mean it's already calculated correctly with chrono crate but you probably know that durations are held only as nanoseconds, so once you've gone to nanoseconds you lose the from datetime and to datetime and therefore it makes it impossible to display correctly.
